src/boosting/gbdt_prediction.cpp

```diff
@@ -44,4 +44,25 @@ void GBDT::PredictLeafIndex(const double* features, double* output) const {
   }
 }
 
+void GBDT::PredictContrib(const double* features, double* output, const PredictionEarlyStopInstance* early_stop) const {
+  int early_stop_round_counter = 0;
+  // set zero
+  const int num_features = max_feature_idx_+1;
+  std::memset(output, 0, sizeof(double) * num_tree_per_iteration_ * (num_features+1));
```

Comment:
there are `max_feature_idx_+1` features, but the explanation also contains the base value as the last element, so the output is `num_features+1` or in other words `max_feature_idx_+2`